runtime/src/main/java/com/dylibso/chicory/runtime/Machine.java

```diff
@@ -131,6 +132,7 @@ static void eval(MStack stack, Instance instance, ArrayDeque<StackFrame> callSta
                         break;
                     case ELSE:
                     case BR:
+                        checkInterruption();
```

Comment:
We don't need to check for interruption after every instruction. As long as we check before branch or calls, the execution should eventually reach a termination point. Although I might be misunderstanding something about WASM.

Reply:
Thanks for the answer!

> As long as we check before branches or calls

This is a good explanation, agree, I suggest that we move this `checkInterruption()` call in the body of `prepareControlTransfer` method so that it takes into account all the branching operations.
All good for the rest, thanks!

Reply:
I considered that, but I think it's unnecessary, as the other usages `BR_IF` and `BR_TABLE` are only forward jumps, and thus we can treat them as any other non-branching instruction. If the execution is only moving forwards, then it will eventually terminate.

While the benchmarks didn't show any performance impact for the checks that I added, they are probably not free, so I tried to only add the where absolutely needed.

Reply:
`BR_TABLE` can be a "backward" jump but it will be immediately followed by a "call" so the reasoning still holds right.
I think we can start with this implementation, can I ask you to capture those tradeoffs in a comment in the code?
Lgtm otherwise, thanks!

Reply:
How does this look? There didn't seem to be a clear place to put the documentation, so I added it to the method.